cmd/soroban-cli/src/config/sign_with.rs

```diff
@@ -0,0 +1,61 @@
+use crate::{signer, xdr::TransactionEnvelope};
+use clap::arg;
+
+use super::{
+    locator,
+    network::{self, Network},
+    secret,
+};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Network(#[from] network::Error),
+    #[error(transparent)]
+    Signer(#[from] signer::Error),
+    #[error(transparent)]
+    Secret(#[from] secret::Error),
+    #[error(transparent)]
+    Locator(#[from] locator::Error),
+    #[error(transparent)]
+    Rpc(#[from] soroban_rpc::Error),
+    #[error("No sign with key provided")]
+    NoSignWithKey,
+    #[error(transparent)]
+    StrKey(#[from] stellar_strkey::DecodeError),
+}
+
+#[derive(Debug, clap::Args, Clone, Default)]
+#[group(skip)]
+pub struct Args {
+    /// Sign with a local key. Can be an identity (--sign-with-key alice), a secret key (--sign-with-key SC36…), or a seed phrase (--sign-with-key "kite urban…"). If using seed phrase, `--hd-path` defaults to the `0` path.
+    #[arg(long, conflicts_with = "sign_with_lab", env = "STELLAR_SIGN_WITH_KEY")]
+    pub sign_with_key: Option<String>,
+    /// Sign with <https://lab.stellar.org>
+    #[arg(
+        long,
+        conflicts_with = "sign_with_key",
+        env = "STELLAR_SIGN_WITH_LAB",
+        hide = true
+    )]
+    pub sign_with_lab: bool,
```

Comment:
also verbose. Should we add `-l` for signing with the lab?

Reply:
As I said it the other comment we can always add this later if it is requested. We could also add an alias `--with-lab`. I do think it's okay to be a little verbose as it makes the scripts more readable and you don't have to go look up the short flags if you forget.